thirdparty/build-thirdparty.sh

```diff
@@ -426,13 +444,14 @@ build_curl() {
     check_if_source_exist $CURL_SOURCE
     cd $TP_SOURCE_DIR/$CURL_SOURCE
 
-    CPPFLAGS="-I${TP_INCLUDE_DIR}" \
-    LDFLAGS="-L${TP_LIB_DIR}" LIBS="-lcrypto -lssl -lcrypto -ldl" \
+    CPPFLAGS="-I${TP_INCLUDE_DIR} -DNGHTTP2_STATICLIB" \
+    LDFLAGS="-L${TP_LIB_DIR}" LIBS="-lcrypto -lssl -lcrypto -ldl -static" \
     CFLAGS="-fPIC" \
+    PKG_CONFIG="pkg-config --static" \
     ./configure --prefix=$TP_INSTALL_DIR --disable-shared --enable-static \
     --without-librtmp --with-ssl=${TP_INSTALL_DIR} --without-libidn2 --disable-ldap --enable-ipv6 \
-    --without-libssh2
-    make -j $PARALLEL && make install
+    --without-libssh2 --without-brotli
```

Comment:
may should add `--without-libpsl` to avoid `undefined reference to psl_free` in some env

Reply:
I think we can refine the thirdparty builds gradually in subsequent PRs. After we encourage hermetic build, there will be a broad range of environment for testing. Corner cases are definitely expected.